### DISCUSSION TITLE
Amazing Disappearing/Reappearing FLAC files...

### DISCUSSION THREAD

**Discussion by awrc** [jcorporation/myMPD#985] [Q&A] [answered]
Hi,

I'm using my mympd with a fairly stripped down mpd (since all my music is in "one file per album" FLAC format and uses cue sheets) and while the track list for a given album shows up exactly as I'd like it to, ie....

01   Track 1 Name
02   Track 2 Name

...and so on, at some point, usually within a few tracks, it starts showing...

⊚  Album Name.flac (where the ⊚ is the little CD image)
01.   Track 1 Name
02.   Track 2 Name

...and so on.  The only way to go back to the default behavior is to delete the database and sticker files.  It might be an MPD issue, rather than a UI thing, since I prefer to keep the album details in .cue files but the .flac files usually have a CUESHEET metadata block, just as a way of getting seek points into the data file.  I've experimented with different playlist plugin combinations in the mpd.conf (by default I don't have any listed) but as often as not that causes all the albums in the database.  Fortunately, at this point I've only got about 20 albums moved to the machine, so it's not a huge problem, but there's the best part of another 1600 to move, and by that time deleted the database every time won't be an option.

Apologies if this turns out to be the case that mympd is doing exactly what it should be, and it's my MPD configuration that's causing the problems.

PS Apologies for garbled track listings - they look OK in the composition window, but go odd when I post the comment.

**Comment by jcorporation** [comment]
It’s likely a MPD issue, but we should validate this. We should grab the mpd protocol responses. Enable verbose output of MPD, replicate the issue, figure out the exact command myMPD sends to MPD and issue this command manually to MPD to get the response.

Please add the command and his response and provide a screenshot.

If this is too complicated for you, please provide the cue and flac files and your mpd version/ settings and I could try to replicate your issue.

**Comment by awrc** [comment]
I think I've found the cause, and it's not really a bug, just unexpected behavior.  Before this particular problem appeared, I'd been going crazy with mute/unmute clicks on tracks on my headphones - a problem that's related to my Raspberry Pi's audio hat, not MPD or myMPD.  What this meant was that I was doing a lot of rebooting of the server and it seems to be that the problem is due to the server rebooting while myMPD/MPD is active.  So here's the listing for an album beforehand:

<img width="1339" alt="Screenshot 2023-02-24 at 1 18 50 PM" src="https://user-images.githubusercontent.com/178767/221275956-f3a410aa-4405-4b9a-ab3b-30224c90e614.png">

Then after that I reboot the server and, once the server is responding again, the same album looks like this:

<img width="1295" alt="Screenshot 2023-02-24 at 1 30 40 PM" src="https://user-images.githubusercontent.com/178767/221276149-f9afa016-a4d8-4ce4-83f7-c79d214d70d8.png">

The relevant sections from the logs immediately after the reboot (up to 13:31:22) and then when I opened a browser window and tried to view the album.

Feb 24 13:31:10 musicbox mpd: update: updating John Foxx/London Overgrown/London Overgrown.flac
Feb 24 13:31:10 musicbox mpd: update: scanning playlist '/opt/music/data/John Foxx/London Overgrown/London Overgrown.cue'
Feb 24 13:31:10 musicbox mpd: update: removing John Foxx/London Overgrown/London Overgrown.cue/track0010
Feb 24 13:31:10 musicbox mpd: update: removing John Foxx/London Overgrown/London Overgrown.cue/track0009
Feb 24 13:31:10 musicbox mpd: update: removing John Foxx/London Overgrown/London Overgrown.cue/track0008
Feb 24 13:31:10 musicbox mpd: update: removing John Foxx/London Overgrown/London Overgrown.cue/track0007
Feb 24 13:31:10 musicbox mpd: update: removing John Foxx/London Overgrown/London Overgrown.cue/track0006
Feb 24 13:31:10 musicbox mpd: update: removing John Foxx/London Overgrown/London Overgrown.cue/track0005
Feb 24 13:31:10 musicbox mpd: update: removing John Foxx/London Overgrown/London Overgrown.cue/track0004
Feb 24 13:31:10 musicbox mpd: update: removing John Foxx/London Overgrown/London Overgrown.cue/track0003
Feb 24 13:31:10 musicbox mpd: update: removing John Foxx/London Overgrown/London Overgrown.cue/track0002
Feb 24 13:31:10 musicbox mpd: update: removing John Foxx/London Overgrown/London Overgrown.cue/track0001
Feb 24 13:31:22 musicbox mpd: client: [3] process command "sticker "list" "song" "John Foxx/London Overgrown/London Overgrown.cue/track0001""
Feb 24 13:31:22 musicbox mpd: client: [3] process command "sticker "list" "song" "John Foxx/London Overgrown/London Overgrown.cue/track0002""
Feb 24 13:31:22 musicbox mpd: client: [3] process command "sticker "list" "song" "John Foxx/London Overgrown/London Overgrown.cue/track0003""
Feb 24 13:31:22 musicbox mpd: client: [3] process command "sticker "list" "song" "John Foxx/London Overgrown/London Overgrown.cue/track0004""
Feb 24 13:31:22 musicbox mpd: client: [3] process command "sticker "list" "song" "John Foxx/London Overgrown/London Overgrown.cue/track0005""
Feb 24 13:31:22 musicbox mpd: client: [3] process command "sticker "list" "song" "John Foxx/London Overgrown/London Overgrown.cue/track0006""
Feb 24 13:31:22 musicbox mpd: client: [3] process command "sticker "list" "song" "John Foxx/London Overgrown/London Overgrown.cue/track0007""
Feb 24 13:31:22 musicbox mpd: client: [3] process command "sticker "list" "song" "John Foxx/London Overgrown/London Overgrown.cue/track0008""
Feb 24 13:31:22 musicbox mpd: client: [3] process command "sticker "list" "song" "John Foxx/London Overgrown/London Overgrown.cue/track0009""
Feb 24 13:31:22 musicbox mpd: client: [3] process command "sticker "list" "song" "John Foxx/London Overgrown/London Overgrown.cue/track0010""
Feb 24 13:42:16 musicbox mpd: client: [0] process command "find "((AlbumArtist == 'John Foxx') AND (Album == 'London Overgrown'))" sort Disc window 0:5000"
Feb 24 13:42:16 musicbox mympd: Sticker for uri "John Foxx/London Overgrown/London Overgrown.flac" not found in cache

So this only seems to happen when there's an unexpected reboot.  Reducing the enabled playlist types to just "m3u" (for mympd's own playlists) and "cue" doesn't get rid of the problem, so it doesn't appear to be caused by confusion between the "cue", "embcue" and "flac" playlists.  In fact, I established that in the case of this particular file, there wasn't an embedded CUESHEET in the FLAC, and adding one in (even though the EMBCUE and FLAC playlist plugins are disabled) resulted in every track appearing twice, once with the correct name from the .cue file, once as track0001, track0002 etc,

Once I finally figure out why the mute/unmute behavior for my Pi's DAC isn't working as it should, reboots should become a very occasional thing, and doing a full rescan of the database after those is just routine maintenance.

**Comment by jcorporation** [comment, marked as answer]
#942

**Comment by awrc** [comment]
Looks like something in the measures suggested in #942 has done the trick.  I stripped all of the FLAC files of all but the unstrippable metadata (i.e. streaminfo), cleaned up the CUE files (not strictly necessary, but stuff bought through Qobuz has rather annoying comments embedded in tracks) than added the stripped down cue files into the CUESHEET tag on the FLAC, and altered the settings so that the only playlist plugins were CUE and M3U (the latter is only there so I can create playlists in myPD).  It appears that the tracks are showing the correct info now.
